### ISSUE TITLE
"Export highlights" ignores highlights from downloaded news articles

### ISSUE THREAD

**Issue by labmem8** [koreader/koreader#10476]
* KOReader version: 2023.04
* Device: kobo libra 2

#### Issue
"Export highlights" functionality doesn't export from downloaded news articles (epub files). I have few notes in downloaded articles via RSS downloader plugin. When I export highlights from all books, it doesn't include notes from those articles. I do export to JSON.
#### Steps to reproduce
1. Go to Top Menu -> News downloader (RSS/Atom) -> Edit news feeds -> Add new feed
2. Enter in URL: https://github.com/koreader/koreader/releases.atom
3. Go back and click Sync news feeds
4. Go to news_folder/Release notes from koreader/
5. Open any downloaded article and highlight something
6. Close files
7. Go to Top Menu -> Export highlights -> Choose formats and services -> Json -> Export to Json
8. Go to Top Menu -> Export highlights -> Export all notes in your library
9. Check freshly made json file in clipboard folder, it doesn't have highlights from downloaded news articles.

**Comment by hius07** [comment]
Yes, because downloaded and opened articles are not saved to History.

See line 1054:
https://github.com/koreader/koreader/blob/d54c5acc11f1d6293dcf4977c73cb7a4101b9069/plugins/newsdownloader.koplugin/main.lua#L1048-L1058

You can close the article, open it again as a book and get highlights exported.

**Comment by labmem8** [comment]
Thank you for answer

Strange, but even moving from news folder doesn't help. 

I tried: 
1. moving outside news folder and exporting all highlights
2. moving, making new highlight already in new location outside news folder, exporting all highlights

both don't work sadly

**Comment by hius07** [comment]
Can you export when the article is opened? ("Export all notes in this book")
Do you see the article in History?

**Comment by labmem8** [comment]
1. Yes, importing manually from every article is possible. 
2. Yes, I see all articles in History menu, including those that are inside news folder.

Unfortunately importing highlights manually from every article is not worth it in my context, because I wanted to use import all highlights for language learning tool i'm making. As a workaround I found, you can also save highlights into vocabulary builder by selecting and clicking dictionary > save to vocab builder, and that way it saves also from news articles into vocabulary_builder.sqlite3. But hopefully it'll get fixed, since it's too much clicks now to save

**Comment by hius07** [comment]
(1) News downloader plugin.
If we go out of the open article via the plugin menu "Go to news folder", the article is removed from the History, as intended.
If we just close the document (go to file browser or open another document), self.download_dir is `nil` (not initialized) and the article is not removed from the History.
Dunno if the plugin author has designed it that way intentionally.

(2) Exporter plugin.
When the document is listed in the History, its highlights must be included to the exported highlights.
Are there other books that are shown in the History and are not exported within "Export all notes in your library"?

**Comment by hius07** [comment]
Needs investigation.
Can you share here the metadata file of the problematic article. It is located in the sdr folder with the same name along the article file.

**Comment by labmem8** [comment]
Here it is, for freshly downloaded article from koreader releases rss feed. Has 2 bookmarks. Is located in news folder: [metadata.txt](https://github.com/koreader/koreader/files/11542756/metadata.txt)
Here is metadata of the same article, moved to another folder: [metadata.txt](https://github.com/koreader/koreader/files/11542819/metadata.txt)

In both cases, it was visible in History. And it didn't import via import all highlights.

**Comment by hius07** [comment]
Thank you.
Unfortunately, cannot reproduce.
Got the article from the KOReader release channel and made "Export all notes in your library".
I see the same highlight as you did, among other books:
```
{
    "author": "N/A",
    "title": "Release KOReader 2023.04 \"Solar Panel\" · koreader/koreader · GitHub",
    "file": "/mnt/us/books/6 TMP/News/Release notes from koreader/23-04-27_23-53_KOReader 2023.04 _Solar Panel_.epub",
    "entries": [
        {
            "time": 1684841504,
            "sort": "highlight",
            "text": "MovableContainer: add support for anchoring initial position (#10230) @poire-z",
            "drawer": "lighten",
            "page": 6,
            "chapter": "Release KOReader 2023.04 \"Solar Panel\" · koreader/koreader · GitHub"
        }
    ]
},
```
KOReader version 2023.04-60

**Comment by labmem8** [comment]
Just checked on Android, and all export works there for exported news test above (rss koreader releases feed). On Kobo it doesnt. Same version v2023.04
